bee-api/bee-rest-api/src/endpoints/mod.rs

```diff
@@ -16,30 +16,35 @@ use storage::StorageBackend;
 
 use crate::types::body::{DefaultErrorResponse, ErrorBody};
 
-use bee_gossip::NetworkCommandSender;
+use bee_gossip::{Keypair, NetworkCommandSender, PeerId};
+use bee_ledger::workers::consensus::{ConsensusWorker, ConsensusWorkerCommand};
 use bee_protocol::workers::{
-    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, PeerManager, PeerManagerResWorker,
-    RequestedMessages,
+    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, MessageSubmitterWorkerEvent, PeerManager,
+    PeerManagerResWorker, RequestedMessages,
 };
 use bee_runtime::{
-    node::{Node, NodeBuilder},
+    event::Bus,
+    node::{Node, NodeBuilder, NodeInfo},
+    resource::ResourceHandle,
     worker::{Error as WorkerError, Worker},
 };
 use bee_tangle::{Tangle, TangleWorker};
 
 use async_trait::async_trait;
 use log::{error, info};
+use tokio::sync::mpsc;
 use warp::{http::StatusCode, Filter, Rejection, Reply};
 
-use bee_ledger::workers::consensus::ConsensusWorker;
-use std::{any::TypeId, convert::Infallible};
+use std::{any::TypeId, convert::Infallible, sync::Arc};
 
 pub(crate) type NetworkId = (String, u64);
 pub(crate) type Bech32Hrp = String;
 
 pub(crate) const CONFIRMED_THRESHOLD: u32 = 5;
 
 pub async fn init<N: Node>(
+    node_id: PeerId,
+    keypair: Keypair,
```

Comment:
This is something that could be done true...
I guess however it would be better to not calculate the id here manually and instead rely on the node. Also the `Local` type in the node stores them separately. So even if the chance is small for an error, maybe keep it like it is